app/controllers/search_controller.rb

```diff
@@ -0,0 +1,6 @@
+class SearchController < ApplicationController
+  def index
+    zip = params['q']
```

Comment:
not completely necessary to set this as a local variable if you are only going to use it once below!